src/Cryptography/Sodium/Bindings/Scrypt.hs

```diff
@@ -0,0 +1,253 @@
+    Ptr CChar ->
+    -- | The length of the password.
+    -- Should be comprised between 'cryptoPWHashScryptSalsa2018SHA256PasswdMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256PasswdMax'.
+    CChar ->
+    -- | The salt, of length 'cryptoPWHashScryptSalsa2018SHA256SaltBytes'.
+    Ptr CUChar ->
+    -- | /opslimit:/ The maximum amount of computations to perform.
+    -- Must be comprised between 'cryptoPWHashScryptSalsa2018SHA256OpsLimitMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256OpsLimitMax'.
+    CULLong ->
+    -- | /memlimit:/ The maximum amount of RAM in bytes that the function will use.
+    -- It is highly recommended to allow the function to use at least 16 MiB.
+    -- This number must be between 'cryptoPWHashScryptSalsa2018SHA256MemLimitMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256MemLimitMax'.
+    CSize ->
+    -- | Returns 0 on success, -1 if the computation didn't complete,
+    -- usually because the operating system refused to allocate the amount of
+    -- requested memory.
+    IO CInt
+
+----------------------
+-- Password Storage --
+----------------------
+
+-- | Generate an C ASCII encoded string which includes:
+--
+-- * The result of a memory-hard, CPU-intensive hash function applied to the password;
+-- * The automatically generated salt used for the previous computation;
+-- * The other parameters required to verify the password: opslimit and memlimit.
+--
+-- /See also:/ [crypto_pwhash_scryptsalsa208sha256_str()](https://doc.libsodium.org/advanced/scrypt#password-storage)
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_pwhash_scryptsalsa208sha256_str"
+  cryptoPWHashScryptSalsa2018SHA256Str ::
+    -- | A pointer to the buffer where the string
+    Ptr CChar ->
+    -- | The password
+    Ptr CChar ->
+    -- | Password length
+    CULLong ->
+    -- | /opslimit:/ The maximum amount of computations to perform.
+    -- The 'cryptoPWHashScryptSalsa2018SHA256OpsLimitInteractive' constant
+    -- is a safe baseline value to use.
+    CULLong ->
+    -- | /memlimit:/ The maximum amount of RAM in bytes that the function will use.
+    -- The 'cryptoPWHashScryptSalsa2018SHA256MemLimitInteractive' constant
+    -- is a safe baseline value to use.
+    CSize ->
+    -- | Returns 0 on success, -1 on error.
+    IO CInt
+
+-- | Verify that the password verification string is valid for the associated password.
+--
+-- /See also:/ [crypto_pwhash_scryptsalsa208sha256_str_verify()](https://doc.libsodium.org/advanced/scrypt#password-storage)
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_pwhash_scryptsalsa208sha256_str_verify"
+  cryptoPWHashScryptSalsa2018SHA256StrVerify ::
+    -- | The password verification string, of size 'cryptoPWHashScryptSalsa2018SHA256StrBytes' bytes and 0-terminated.
+    Ptr CChar ->
+    -- | The password.
+    Ptr CChar ->
+    -- | The password length.
+    CULLong ->
+    -- | Returns 0 on success, -1 on error.
```

Comment:
Does 'error' here mean 'password verification failed' or 'help my OS is on fire'?